BondageClub/Scripts/Speech.js

```diff
@@ -94,360 +89,1630 @@ function SpeechGarble(C, CD, NoDeaf) {
 		else if (Player.GetDeafLevel() >= 1) GagEffect = Math.max(GagEffect, 2);
 	}
 
-	if (GagEffect > 0) NS = SpeechGarbleByGagLevel(GagEffect, CD);
+	return GagEffect;
+}
+
+/**
+ * Processes the character's speech, anything between parentheses isn't touched. Effects alter the speech differently according to a character's language. Effects that can be applied are the following: gag talk, baby talk and stuttering.
+ * @param {Character} C - The character, whose dialog might need to be altered
+ * @param {string} CD - The character's dialog to alter
+ * @returns {string} - Returns the dialog after speech effects were processed (Garbling, Stuttering, Baby talk)
+ */
+function SpeechGarble(C, CD, NoDeaf) {
+
+	let GagEffect = 0;
+	let NS = CD;
+
+	// The usual preamble - calculate the gag effect level, and the character's deafness level once
+	GagEffect = SpeechGetGagEffect(C, NoDeaf);
 
-	// No gag effect, we return the regular text
+	// If GagEffect is 0 then there is no need to run this part so, a simple if statement is used in order to check.
+	if (GagEffect > 0) {
+
+		NS = SpeechGarbleByGagLevel(GagEffect, CD);
+	}
 	NS = SpeechStutter(C, NS);
 	NS = SpeechBabyTalk(C, NS);
 
 	return NS;
+
+}
+
+/**
+ * Chooses the correct function to run via a switch statement and returns it when found.
+ * @param {number} GagEffect - The gag effect level that is used to determine the correct garbling function.
+ * @returns {string} - Returns the correct garbling function to run.
+ */
+function SpeechGetGarblingFunction(GagEffect) {
+	
+	// Makes sure that the gag level is within acceptable parameters. As a GagEffect of more than 20 does nothing, but for this function above 20 will break it.
+	if (GagEffect > 20) GagEffect = 20;
+
+	switch (GagEffect) {
+		case 20:
+			return SpeechGarble20;
+		case 19:
+		case 18:
+		case 17:
+		case 16:
+			return SpeechGarble16;
+		case 15:
+		case 14:
+		case 13:
+		case 12:
+			return SpeechGarble12;
+		case 11:
+		case 10:
+		case 9:
+		case 8:
+			return SpeechGarble8;
+		case 7:
+			return SpeechGarble7;
+		case 6:
+			return SpeechGarble6;
+		case 5:
+			return SpeechGarble5;
+		case 4:
+			return SpeechGarble4;
+		case 3:
+			return SpeechGarble3;
+		case 2:
+			return SpeechGarble2;
+		case 1:
+			return SpeechGarble1;
+	}
 }
 
 /**
- * The core of the speech garble function, usable without being tied to a specific character
+ * Garbles a line of text based on the provided gag effect level.
  * @param {Int} GagEffect - The gag level of the speech
  * @param {string} CD - The character's dialog to alter
  */
 function SpeechGarbleByGagLevel(GagEffect, CD, IgnoreOOC) {
+    const garblingFunction = SpeechGetGarblingFunction(GagEffect);
+    return SpeechGarbleLine(CD, garblingFunction, IgnoreOOC);
```

Comment:
```suggestion
	if (GagEffect > 0) {
		const garblingFunction = SpeechGetGarblingFunction(GagEffect);
		return SpeechGarbleLine(CD, garblingFunction, IgnoreOOC);
	} else {
		return CD;
	}
```
One final suggestion. It's not an issue now, but since `SpeechGetGarblingFunction` will return `undefined` for a gag effect of 0, calling `SpeechGarbleByGagLevel(0, someText)` would throw an error. This would make the function safer and a bit more flexible for consumers with no real downside (it also means you'd be able to remove the `if` clause on line 110).

Reply:
Nice catch Ellie I forgot about that. I remembered it when writing up the inital function though so :)